docs/feeds.md

```diff
@@ -0,0 +1,106 @@
+
+
+With this method there is _no need_ to stand up any persistent service outside of OpenWhisk.  All feed management happens naturally though stateless OpenWhisk *feed actions*, which negotiate
+directly with a third part webhook API.
+
+When invoked with `CREATE`, the feed action simply installs a webhook for some other service, asking the remote service to POST notifications to the appropriate `fireTrigger` URL in OpenWhisk.
+
+The webhook should be directed to send notifications to a URL such as:
+`POST /namespaces/{namespace}/triggers/{triggerName}`
+
+# Implementing Feeds with Polling
+
+It is possible to set up an OpenWhisk *action* to poll a feed source entirely within OpenWhisk, without the need to stand up any persistent connections or external service.
+
+For feeds where a webhook is not available, but do not need high-volume or low latency response times, polling is an attractive option.
+
+To set up a polling-based feed, the feed action takes the following steps when called for `CREATE`:
+1.   The feed action sets up a periodic trigger (*T*) with the desired frequency, using the `whisk.system/alarms` feed.
+2.   The feed developer creates a `pollMyService' action which simply polls the remote service and returns any new events.
+3.  The feed action sets up a *rule*
+*T -> pollMyService*.
+
+That's it.   This procedure implements a polling-based trigger entirely using OpenWhisk actions, without any need for a separate service.
+
+# Implementing Feeds via Connections
+
+The previous 2 architectural choices are simple and easy to implement. However, if you want a high-performance feed, there is no substitute for persistent connections and long-polling or similar techniques.
+
+Since OpenWhisk actions are stateless,  an action cannot maintain a persistent connection to a third party. Instead, we are forced to stand up a separate service (outside of OpenWhisk) that runs all the time.   We call these *provider services*.  A provider service can maintain connections to third party event sources that support long polling or other connection-based notifications.   
+The provider service should provide a REST API that allows the OpenWhisk *feed action* to control the feed.   The provider service acts as a proxy between the event provider and OpenWhisk -- when it receives events from the third party, it sends them on to OpenWhisk by firing a trigger.
+
+The Cloudant *changes* feed is the canonical example -- it stands up a `cloudanttrigger` service which mediates between Cloudant notifications over a persistent connection, and OpenWhisk triggers.
+<!-- TODO: add a reference to the open source implementation -->
+
+The *alarm* feed is implemented with a similar pattern.
+
+The connection-based architecture is the highest performance option -- but it's far more difficult to operate and maintain than the polling and hook architectures.   The provider service must be production-quality: it must be highly-available and fault-tolerant.    
```

Comment:
This sentence has too much internal info that user doesn't care. 

The connection-based architecture is the highest performance option.
We already established at the beginning of the section that was the high performance feed.
So no need to repeat here.
 
-- but it's far more complex to operate and maintain than the polling and hook architectures   The provider service must be production-quality: it must be highly-available and fault-tolerant. 

Everything needs to be production-quality so not much explanation why is called out here. 
You can say the same as webhooks, if they don't have a reliable way or hihg performance way to call webhooks they will have problems.

Maybe what you want to tell users  that if your are going to use a third party service that uses the connection-based architecture or you going to implemented on your own that for large amount of traffic you need to take into consideration performance and throttle implications to have a high quality event stream into OpenWhisk.